app/controllers/media_objects_controller.rb

```diff
@@ -318,9 +318,10 @@ def custom_update
   end
 
   def index
+    mos = MediaObject.accessible_by(current_ability, :index)
     respond_to do |format|
       format.json {
-        paginate json: MediaObject.accessible_by(current_ability, :index)
+        paginate json: mos.to_a.collect { |mo| mo.as_json(include_structure: params[:include_structure] == "true") }
```

Comment:
Does the paginate method need to be moved up with the `accessible_by` relation to match Admin::CollectionsController?  I'm concerned that this might be accidentally loading more than it needs to without this change which could impact performance at scale.  I'm not certain at all but it is a guess.

As far as I recall `paginate` should add limit and paging filters on the solr query.  What I'm not sure about is if `mos.to_a` will execute before or after adding those filters.  I'm not sure what the best way to test this would be but manually stepping through the code might be a good enough effort for now.